lib/vtls/schannel.h

```diff
@@ -36,6 +36,15 @@
 #ifdef _MSC_VER
 #pragma warning(pop)
 #endif
+
+#if !defined(_WIN32_WINNT) || _WIN32_WINNT < 0x0600
+#undef _WIN32_WINNT
+/* Define _WIN32_WINNT to ensure MinGW headers show NCrypt definitions.
+ * MinGW 4.8.1 headers require _WIN32_WINNT>=600 to show  NCrypt symbols.
+ * curl documents Vista as lowest supported Windows version */
+#define _WIN32_WINNT 0x0600  /* Requires Windows Vista */
```

Comment:
curl currently supports XP as minimum.
Bumping to Vista is scheduled for later via #18009.
A logic like above will need to be done in the build
systems (and config-win32.h), or in the env.